packages/fast-layouts-react/src/page/page.tsx

```diff
@@ -48,27 +41,18 @@ export class Page extends Foundation<PageHandledProps, PageUnhandledProps, {}> {
     }
 
     private generateAttributes(): React.HTMLAttributes<HTMLDivElement> {
-        const attributes: React.HTMLAttributes<HTMLDivElement> = Object.assign(
-            {},
-            this.unhandledProps(),
-            {
-                className: super.generateClassNames(this.props.managedClasses.page),
-            }
-        );
         const columns: string = `${this.props.margin} minmax(0, ${this.props.maxWidth}) ${
             this.props.margin
         }`;
 
-        if (!attributes.style) {
-            attributes.style = {};
-        }
-
-        attributes.style = {
-            gridTemplateColumns: columns,
-            msGridColumns: columns,
+        return {
+            ...this.unhandledProps(),
+            className: super.generateClassNames(this.props.managedClasses.page),
+            style: {
+                gridTemplateColumns: columns,
+                msGridColumns: columns,
+            },
```

Comment:
Will pre-existing style attributes here return a combined style object, or is this going to blow away the entire style object due to shallow compare of the spread operator?

Reply:
No, it won't merge the style object, but the current code doesn't do that either.
Note how it always overwrites `attributes.style` with the new object (line 66), throwing away what was there before.
That effectively means that the if block on line 62 is useless.

If we wanted to merge the properties, we would have to make another spread on style.

Is that the intended behavior?

Reply:
Ah - yes, looks like that was likely a bug; I think that would be the intended behavior, otherwise unhandled style props wouldn't be honored. Do you want to include that change as part of this?

Reply:
I can.
What should take precedence on style? The unhandled props or this?
I mean, should we allow styles.gridTemplateColumns to be overridden by unhandledProps?
I assume no.

Reply:
Yep - I agree with that assessment